tests/test_query.py

```diff
@@ -108,6 +109,79 @@ def test_app_with_set_allowed_origins(
+            "last_updated": 1689609836,
+            "name": "Generalized Self-Efficacy Scale",
+            "definition_text": "The original Generalized Self-Efficacy Scale contains 10 items designed to tap into a global sense of self-efficacy, or belief of an individual in his or her ability (e.g., \u201cI can always solve difficult problems if I try hard enough,\u201d and \u201cI can usually handle whatever comes my way.\u201d) The revised version here includes these 10 items and two, which are repeated and reversed to examine acquiescence bias. Response options range from 1, never true, to 7, always true. Higher scores indicate greater generalized self-efficacy.",
+            "id": "tsk_p7cabUkVvQPBS",
+        },
+        {
+            "creation_time": 1689610375,
+            "last_updated": 1689610375,
+            "name": "Verbal Interference Test",
+            "definition_text": "The Verbal Interference Test is a behavioral assessment of cognitive regulation. In this task participants are presented with visual word stimuli that appear with incongruent text and color meaning (e.g., the word \u201cRED\u201d printed in blue, the word \u201cBLUE\u201d printed in green, the word \u201cGREEN\u201d printed in red). There are two phases of the task: Name (Part I) and Color (Part II). In the Name phase, participants are asked to identify the meaning of the word (e.g., red is the correct answer for the word \u201cRED\u201d printed in blue). In the Color phase, participants are asked to identify the color in which the word is printed (e.g., blue is the correct answer for the word \u201cRED\u201d printed in blue). This test assesses aspects of inhibition and interference corresponding to those indexed by the Stroop test.",
+            "id": "tsk_ccTKYnmv7tOZY",
+        },
+    ]
+
+    def mock_httpx_get(**kwargs):
+        return httpx.Response(status_code=200, json=mock_vocab_json)
+
+    monkeypatch.setattr(httpx, "get", mock_httpx_get)
+
+    with test_app:
+        term_labels_path = (
+            test_app.app.state.vocab_dir_path
+            / "cogatlas_task_term_labels.json"
+        )
+        assert term_labels_path.exists()
+
+        with open(term_labels_path, "r") as f:
+            term_labels = json.load(f)
+
+        assert term_labels == {
+            "tsk_p7cabUkVvQPBS": "Generalized Self-Efficacy Scale",
+            "tsk_ccTKYnmv7tOZY": "Verbal Interference Test",
+        }
+
+
+def test_failed_vocab_fetching_on_startup_raises_warning(
+    test_app, monkeypatch
+):
+    """Tests that when a GET request to the Cognitive Atlas API fails, a warning is raised and that a term label lookup file is still created using a backup copy of the vocab."""
+    monkeypatch.setenv(util.GRAPH_USERNAME.name, "SomeUser")
+    monkeypatch.setenv(util.GRAPH_PASSWORD.name, "SomePassword")
+
+    def mock_httpx_get(**kwargs):
+        return httpx.Response(
+            status_code=500, json={}, text="Some error message"
```

Comment:
Hmm, my understanding was that `4xx` codes are typically "client-side" errors, and `5xx` are server-side errors. In this test case, I was envisioning a scenario where the the request fails because the Cognitive Atlas API itself is down for some reason - which would be server-side? 

From what I recall of the errors we would get when our own API was down, and looking at the httpx status codes
(e.g., https://github.com/encode/httpx/blob/fdaa01275a1b80e54be4423e579f9a19f7c63d8f/httpx/status_codes.py#L124C1-L125), it looks like maybe a 503 code specifically would be more appropriate example here:
```
SERVICE_UNAVAILABLE = 503, "Service Unavailable"
```

I've updated the test and docstring to clarify the test case - let me know if this doesn't make sense to you.

Reply:
ah, I see. Thanks for explaining, that makes sense to me